RevenueCatUI/CustomerCenter/ViewModels/CustomerCenterViewModel.swift

```diff
@@ -165,27 +172,52 @@ private extension CustomerCenterViewModel {
 
     func loadPurchaseInformation(customerInfo: CustomerInfo) async throws {
         let hasActiveProducts =  !customerInfo.activeSubscriptions.isEmpty ||
-        !customerInfo.nonSubscriptions.isEmpty
+            !customerInfo.nonSubscriptions.isEmpty
 
         if !hasActiveProducts {
-            self.purchaseInformation = nil
+            self.activePurchases = []
             self.state = .success
             return
         }
 
-        guard let activeTransaction = customerInfo.earliestExpiringTransaction() else {
+        var activePurchases: [PurchaseInformation] = []
+
+        for subscription in customerInfo.activeSubscriptions
+            .compactMap({ id in customerInfo.subscriptionsByProductIdentifier[id] })
+            .sorted(by: {
+                guard let date1 = $0.expiresDate, let date2 = $1.expiresDate else {
+                    return $0.expiresDate != nil
+                }
+
+                return date1 < date2
+            }) {
+
+            let entitlement = customerInfo.entitlements.all.values
+                .first(where: { $0.productIdentifier == subscription.productIdentifier })
+
+            let purchaseInfo = try await createPurchaseInformation(
+                for: subscription,
+                entitlement: entitlement,
+                customerInfo: customerInfo)
+
+            activePurchases.append(purchaseInfo)
+        }
+
+        self.activePurchases = activePurchases
+        if let activeTransaction = customerInfo.earliestExpiringTransaction() {
+            // get the active non-subscription transaction
+            let entitlement = customerInfo.entitlements.all.values
+                .first(where: { $0.productIdentifier == activeTransaction.productIdentifier })
+
+            self.purchaseInformation = try await createPurchaseInformation(
+                for: activeTransaction,
+                entitlement: entitlement,
+                customerInfo: customerInfo
+            )
+        } else if activePurchases.isEmpty {
             Logger.warning(Strings.could_not_find_subscription_information)
-            self.purchaseInformation = nil
             throw CustomerCenterError.couldNotFindSubscriptionInformation
         }
-
-        let entitlement = customerInfo.entitlements.all.values
-            .first(where: { $0.productIdentifier == activeTransaction.productIdentifier })
-
-        self.purchaseInformation = try await createPurchaseInformation(
-            for: activeTransaction,
-            entitlement: entitlement,
-            customerInfo: customerInfo)
```

Comment:
Taking this out of here makes it so that if none of the two `if`s above are true, this won't be set. Is it intentional?

Reply:
Addressed in https://github.com/RevenueCat/purchases-ios/pull/5089/commits/2737508985f467232a073ae367e78da1a1c4f40f